File: Sorts/python/selection_sort.py
def selection_sort(list):
    for i in range(len(list)):
        minIndex = i
        j = i + 1
        
        while j <= len(list) - 1:
            if list[j] < list[minIndex]:
                minIndex = j
            j += 1
        
        temp = list[minIndex]
        list[minIndex] = list[i]
        list[i] = temp
    return list

list = [3, 5, 6, 7, 8, 12, 1, 3, 4]

print(selection_sort(list))

File: Sorts/js/selection_sort.js
function selection_sort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let minIndex = i; // init min
    for (let j = i; j < arr.length; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
        isMinUpdated = true;
      }
    }
    if (minIndex !== i) {
      swap(arr, i, minIndex);
    }
  }

  return arr;
}

function swap(arr, indexOne, indexTwo) {
  let temp = arr[indexOne];
  arr[indexOne] = arr[indexTwo];
  arr[indexTwo] = temp;
}

console.log(selection_sort([4, 6, 1, 15, 13, 11, 9, 7]));
